pkgs/build-support/setup-hooks/patch-dylib.sh

```diff
@@ -0,0 +1,55 @@
+# The equivalent of patchelf but for .dylib / mach-o / shared libraries on macOS
+#
+# Patches the paths to shared library dependencies to use libs in the nix store.
+# The script is not magical and expects the caller to provide a list of paths
+# where library dependencies can be found.
+#
+# For example if a.dylib depended on /usr/lib/b.dylib, there's a chance the latter
+# does not actually exist and can be found somewhere in /nix/store/XXX-libb/lib/b.dylib.
+#
+# Dependencies:
+#  - ripgrep
+#  - sed (probably in the gnused package)
+#
+# Arguments:
+#  1. targetLib: path to a shared library that has to be patched
+#  2. libraryPaths: <colon> (:) separated list of paths
+function patchDylib(){
+  local targetLib="$1"
+  local libraryPaths="$2"
+
+  # Make the shared paths absolute
+  install_name_tool -id $targetLib $targetLib
+  # Get all dependent shared libs
+  libs="$(otool -L $targetLib \
+          | rg -N '^\s+(?P<lib>.+)\s\(.+\)' -r '$lib' \
```

Comment:
`-N` is `--no-line-number`. I'll use long-style options in the future. Is there a check-list for these kinds of style decisions?

Reply:
I know what it does, checked the man page. It's just that we're not running it from an interactive shell in a terminal so Ripgrep wouldn't include line numbers. The fewer options you need to mentally parse the easier it is to see what a command's doing.

Don't think there is, consider it my personal preference. If you'd used short options everywhere I wouldn't have mentioned it but you mixed the two. Consistency means you'd only have to look up an option once. The preference for long options is just because these will more frequently be read by someone trying to understand this code than written, since it's a script.